contrib/hashicorp/consul/consul.go

```diff
@@ -80,6 +99,7 @@ func (k *KV) startSpan(resourceName string, key string) ddtrace.Span {
 		tracer.Tag(ext.Component, componentName),
 		tracer.Tag(ext.SpanKind, ext.SpanKindClient),
 		tracer.Tag(ext.DBSystem, ext.DBSystemConsulKV),
+		tracer.Tag(ext.NetworkDestinationName, k.consulConfig.hostname),
```

Comment:
I believe this will panic if `WrapClient` is used, which ideally would have been caught by the tests, but it looks like we don't have any for `WrapClient.`

Reply:
Pending our discussion above, if we continue using the new wrapper function, I will write new tests to differentiate because the old wrapper should not set this tag and vice versa